extensions/wikia/Search/classes/Services/ESFandomSearchService.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Wikia\Search\Services;
+
+class ESFandomSearchService extends AbstractSearchService {
+	const RESULTS_COUNT = 6;
+
+	protected function prepareQuery( $phrase ) {
+		return urlencode( $phrase );
+	}
+
+	protected function select( $query ) {
+		global $wgConsulServiceTag, $wgConsulUrl;
+
+		$consulUrl =
+			( new \Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl,
+				$wgConsulServiceTag ) )->getUrl( 'fandom-search' );
+
+		$response = \Http::post( "http://$consulUrl/fandom", array(
+			'noProxy' => true,
+			'postData' => "queryTerms=$query",
+			'headers' => [
+				'Content-Type' => 'application/x-www-form-urlencoded',
+			],
+		) );
+		if ( $response !== false ) {
+			$decodedResponse = json_decode( $response, true );
+			if ( json_last_error() !== JSON_ERROR_NONE ) {
+				\WikiaLogger::instance()->error( " Fandom Stories Search: error decoding response", [
+					'query' => $query,
+					'response' => $response,
+					'error' => json_last_error(),
+				] );
+			} else {
+				if ( isset( $decodedResponse['hits']['hits'] ) ) {
```

Comment:
minor: turning `else` into `else if` on line 34 seems simpler and reduces level of nesting